File: 파이썬/다리를지나는트럭.py
from collections import deque


def solution(bridge_length, weight, truck_weights):
    trucks = deque(truck_weights)
    NONE = 0
    bridge = deque([NONE for _ in range(bridge_length)])
    time = 0

    while trucks or any(bridge):
        bridge.popleft()
        time += 1
        if trucks and sum(bridge) + trucks[0] <= weight:
            bridge.append(trucks.popleft())
        else:
            bridge.append(NONE)
    return time

File: 자바스크립트/다리를지나는트럭.js
function solution(bridge_length, weight, truck_weights) {
    const NONE = 0;
    let bridge = Array.from(new Array(bridge_length), () => NONE);
    let time = 0;
    let bridgeWeight = 0;

    while (truck_weights.length > 0 || bridge.some(x => x !== 0)) {
        bridgeWeight -= bridge.shift();
        time++;
        const newItem = (bridgeWeight + truck_weights[0] <= weight) ? truck_weights.shift() : NONE;
        bridge.push(newItem)
        bridgeWeight += newItem
    }
    return time;
}